src/stk/calculators/calculator.py

```diff
@@ -0,0 +1,84 @@
+"""
+Calculator
+==========
+
+
+
+"""
+
+
+class Calculator:
+    def __init__(self, use_cache):
+        self._use_cache = use_cache
+        self._cache = {}
+
+    def set_cache_use(self, use_cache):
+        """
+        Set cache use on or off.
+
+        Parameters
+        ----------
+        use_cache : :class:`bool`
+            ``True`` if the cache is to be used.
+
+        Returns
+        -------
+        :class:`.Calculator`
+            The calculator.
+
+        """
+
+        self._use_cache = use_cache
+
+    def is_caching(self):
+        """
+        ``True`` if the calculator has caching turned on.
+
+        Returns
+        -------
+        :class:`bool`
+            ``True`` if the calculator has caching turned on.
+
+        """
+
+        return self._use_cache
+
+    def add_to_cache(self, mol, value=None):
+        """
+        Add a molecule to the cache.
+
+        Parameters
+        ----------
+        mol : :class:`.Molecule`
+            The molecule to be added to the cache.
+
+        value : class:`object`, optional
+            The cached value associated with the molecule.
+
+        Returns
+        -------
+        :class:`.Calculator`
+            The calculator.
+
+        """
+
+        self._cache.add(mol)
+        return self
+
+    def is_in_cache(self, mol):
```

Comment:
Same as above comment for naming. Maybe name could be ```check_in_cache``` or even ```check_mol_in_cache```, but this may be excessive.

Reply:
Again, this is my own opinion and is subjective.